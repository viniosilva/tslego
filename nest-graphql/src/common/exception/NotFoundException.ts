export class NotFoundException extends Error {
  constructor(readonly message: string) {
    super(message);
  }
}
