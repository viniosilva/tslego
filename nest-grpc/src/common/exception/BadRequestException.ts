export class BadRequestException extends Error {
  constructor(readonly message: string) {
    super(message);
  }
}
