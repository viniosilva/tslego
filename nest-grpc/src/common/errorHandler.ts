import { RpcException } from '@nestjs/microservices';
import { ServiceError, status } from 'grpc';
import { NotFoundException } from './exception/NotFoundException';
import { logger } from './logger';

export function errorHandler(err: Error): void {
  if (err instanceof NotFoundException) {
    throw new RpcException({
      code: status.NOT_FOUND,
      message: err.message,
    } as ServiceError);
  } else {
    logger.child({ stackError: err.stack }).error(err.message);
    throw new RpcException({
      code: status.INTERNAL,
      message: 'Internal server error',
    } as ServiceError);
  }
}
