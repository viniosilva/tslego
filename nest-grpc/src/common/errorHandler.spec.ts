import { errorHandler } from './errorHandler';
import { NotFoundException } from './exception/NotFoundException';

describe('common', () => {
  describe('errorHandler', () => {
    it('should throw not found', () => {
      const fn = () => errorHandler(new NotFoundException('NOT_FOUND'));
      expect(fn).toThrowErrorMatchingSnapshot();
    });

    it('should throw internal server error', () => {
      const fn = () => errorHandler(new Error('INTERNAL SERVER ERROR'));
      expect(fn).toThrowErrorMatchingSnapshot();
    });
  });
});
