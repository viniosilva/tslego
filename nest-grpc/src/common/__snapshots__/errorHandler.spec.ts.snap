// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common errorHandler should throw internal server error 1`] = `"Internal server error"`;

exports[`common errorHandler should throw not found 1`] = `"NOT_FOUND"`;
